src/Installer/dnup/NonUpdatingProgressTarget.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection.Metadata.Ecma335;
+using System.Text;
+
+namespace Microsoft.DotNet.Tools.Bootstrapper;
+
+public class NonUpdatingProgressTarget : IProgressTarget
+{
+    public IProgressReporter CreateProgressReporter() => new Reporter();
+
+    class Reporter : IProgressReporter
+    {
+        List<ProgressTaskImpl> _tasks = new();
+
+        public IProgressTask AddTask(string description, double maxValue)
+        {
+            var task = new ProgressTaskImpl(description)
+            {
+                MaxValue = maxValue
+            };
+            _tasks.Add(task);
+            Spectre.Console.AnsiConsole.WriteLine(description + "...");
```

Comment:
nit: Should we add ... to the description every time? And should we abstract spectre.console out of the class? I don't think we need to do either for now.